src/workerd/api/trace.h

```diff
@@ -337,6 +334,26 @@ class TraceDiagnosticChannelEvent final: public jsg::Object {
   kj::Array<kj::byte> message;
 };
 
+class ScriptVersion final: public jsg::Object {
```

Comment:
I suspected a "resource type" `jsg::Object` wasn't needed here but wasn't sure what the trade offs were and what the right alternative was.

What are the key tradeoffs/differences between the 3 options here of `jsg::Object`, `JSG_STRUCT` and `jsg::Dict`.

E.g.
* When `JSG_STRUCT` fields are accessed in JS, are they always copied?
* `jsg::Dict` only supports a single value type for its fields - might we want to avoid if we think there's a chance we want to add a non-string field value in the future, or could we change it later without breaking users?
* What are the concrete disadvantages of using `jsg::Object` here? The fact it's more boilerplate in C++ code? The fact that it simply implies it's more than just a dumb data type to the reader of the code? Are extra operations available in JS on `jsg::Object`s that aren't available on structs?

I'm leaning towards using `JSG_STRUCT` and will make those changes but interested in what the implications are.

Reply:
@jasnell I've changed this to a `JSG_STRUCT` but my initial implementation creates a new instance of this struct and thus copies its string members each time `trace.scriptVersion` is accessed. Is there a way around this?

I wondered if `MemoizedIdentity` would help but I can't seem to get that to work.